widgetry/src/style/buttons.rs

```diff
@@ -224,6 +224,55 @@ impl<'a> StyledButtons<'a> for Style {
     }
 }
 
+impl<'a> ButtonStyle {
+    pub fn plain(&self) -> ButtonBuilder<'a> {
+        basic_button(self, None)
+    }
+
+    pub fn outline(&self) -> ButtonBuilder<'a> {
+        self.plain()
+            .outline(self.outline_thickness, self.outline, ControlState::Default)
+    }
+
+    pub fn plain_text(&self, text: &'a str) -> ButtonBuilder<'a> {
+        self.plain().label_text(text)
+    }
+
+    pub fn plain_icon(&self, image_path: &'a str) -> ButtonBuilder<'a> {
+        icon_button(self.plain().image_path(image_path))
+    }
+
+    pub fn plain_icon_bytes(&self, labeled_bytes: (&'a str, &'a [u8])) -> ButtonBuilder<'a> {
+        icon_button(self.plain().image_bytes(labeled_bytes))
+    }
+
+    pub fn plain_icon_text(&self, image_path: &'a str, text: &'a str) -> ButtonBuilder<'a> {
+        self.plain()
+            .label_text(text)
+            .image_path(image_path)
+            .image_dims(ScreenDims::square(18.0))
+    }
+
+    pub fn outline_text(&self, text: &'a str) -> ButtonBuilder<'a> {
```

Comment:
For ease of sharing styles with pregame I've slightly rephrased some methods from StyledButtons (which is a trait that EvnetCtx implements) onto ButtonStyle. 

I'm actually going to attempt a refactor based on this to try to get rid of the cartesian explosion of methods on StyledButtons. The result will be less method definitions, but I'm not yet sure if the resultant API will be better. I'll have a go at it and you can let me know what you think.

Reply:
Sounds like a promising refactor. The levels of indirection here definitely make it a bit tedious to figure out what's going on.